QATCH/VisQAI/src/models/predictor.py

```
@@ -319,18 +320,28 @@ def update(
+            comment = self._calculate_sha256_safe(zip_path)
+            password = hashlib.sha256(comment.encode()).hexdigest()
+            # replace ZIP with encrypted version
+            self._add_password_and_comment(
+                zip_path, zip_path, password, comment)
+            successful = True
+            Log.i("Updated ensemble saved to archive")
         except Exception as e:
             Log.e(f"Error saving updated ensemble: {e}")
-            raise RuntimeError(f"Failed to save updated ensemble: {e}")
+        finally:
+            return successful
```

Ruff (0.12.2)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)